File: LibraryApp/Ee.Ebs.Application.Contracts/Books/DTOs/BookDto.cs
namespace Ee.Ebs.Application.Contracts.Books.DTOs;

public class BookDto
{
    public int Id { get; set; }
    public bool IsActive { get; set; } 
    public string Title { get; set; }
    public int PublishYear { get; set; }
    public bool IsBorrowed { get; set; }
        
    // Navigation property'lerden AutoMapper ile doldurulacak alanlar
    public string AuthorName { get; set; }
    public string CategoryName { get; set; }
    public string PublisherName { get; set; }
}

File: angular/src/app/features/books/models/book.dto.ts
export interface BookDto {
  id: number;
  isActive: boolean;
  title: string;
  publishYear: number;
  isBorrowed: boolean;
  authorName: string;
  categoryName: string;
  publisherName: string;
}

export interface BookCreateDto {
  authorID: number;
  title: string;
  publishYear: number;
  categoryID: number; 
  publisherId: number;
  authorFullName: string;
}

export interface BookEditDto {
  id: number;
  authorID: number;
  title: string;
  publishYear: number;
  isBorrowed: boolean;
  publisherId: number;
  categoryID: number;
}